scripts/release.py

```diff
@@ -206,16 +206,237 @@ def create_draft_release(
         )
-        print("Please review your release notes draft here: https://github.com/DataDog/dd-trace-py/releases")
+        print("\nPlease review your release notes draft here: https://github.com/DataDog/dd-trace-py/releases")
+
+    return name, rn
+
+
+def setup_gh():
+    # get dd-trace-py repo
+    gh_token = os.getenv("GH_TOKEN")
+    if not gh_token:
+        raise ValueError(
+            "We need a Github token to generate the release notes. Please follow the instructions in the script."
+        )
+    g = Github(gh_token)
+    return g.get_repo(full_name_or_id="DataDog/dd-trace-py")
+
+
+def create_notebook(dd_repo, name, rn, base, rc, patch):
+    dd_api_key = os.getenv("DD_API_KEY_STAGING")
+    dd_app_key = os.getenv("DD_APP_KEY_STAGING")
+    if rc:
+        # need to figure out which versions to compare in order to get commits
+        # if this is the first RC need to grab the last released version branch e.g base = 1.19, compare = 1.18
+        if int(name[-1]) == 1:
+            last_version = base[:-1] + str(int(base[-1]) - 1)
+        # this script does not currently create notebooks for anything other than RC1 versions
+        else:
+            print(
+                "Since this is not the RC1 for this release."
+                "Please add the release notes for this release to the notebook."
+            )
+            return
+
+    diff_raw = subprocess.check_output(
+        "git cherry -v {last_version} 1.x".format(last_version=last_version),
+        shell=True,
+        cwd=os.pardir,
+    ).decode("utf8")
+
+    commit_hashes = re.findall("[0-9a-f]{5,40}", diff_raw)
+    commits = []
+    # get the commit objects
+    for commit_hash in commit_hashes:
+        commits.append(dd_repo.get_commit(commit_hash))
+
+    # get list of authors for when we make the slack announcement
+    author_slack_handles = []
+    prs_details = []
+    # from each commit:
+    # 1. get release note text
+    # 2. pull out author and turn into email
+    # 3. get PR number and create link
+    for commit in commits:
+        files = commit.files
+        for file in files:
+            filename = file.filename
+            if "releasenotes/notes/" in filename:
+                # we need to make another api call to get ContentFile object so we can see what's in there
+                rn_file_content = dd_repo.get_contents(filename).decoded_content.decode("utf8")
+                # try to grab a good portion of the release note for us to use to insert in our reno release notes
+                # this is a bit hacky, will only attach to one section if you have multiple sections in a release note
+                # (e.g. a features and a fix section):
+                # for example: https://github.com/DataDog/dd-trace-py/blob/1.x/releasenotes/notes/asm-user-id-blocking-5048b1cef07c80fd.yaml # noqa
+                try:
+                    rn_piece = re.findall(
```

Comment:
why are you using regular expression? these are yaml, can't we just parse the yaml?

Reply:
Wow, that might be way nicer, backlog it is.